pkg/store/pruning.go

```diff
@@ -0,0 +1,67 @@
+package store
+
+import (
+	"context"
+
+	ds "github.com/ipfs/go-datastore"
+
+	"github.com/rollkit/rollkit/pkg/config"
+	"github.com/rollkit/rollkit/types"
+)
+
+// DefaultPruningStore is for a store that supports pruning of block data.
+type DefaultPruningStore struct {
+	Store
+
+	config                config.PruningConfig
+	latestBlockDataHeight uint64
+}
+
+var _ PruningStore = &DefaultPruningStore{}
+
+func NewDefaultPruningStore(ds ds.Batching, config config.PruningConfig) PruningStore {
+	return &DefaultPruningStore{
+		Store: &DefaultStore{db: ds},
+
+		config: config,
+	}
+}
+
+// SaveBlockData saves the block data and updates the latest block data height.
+func (s *DefaultPruningStore) SaveBlockData(ctx context.Context, header *types.SignedHeader, data *types.Data, signature *types.Signature) error {
+	err := s.SaveBlockData(ctx, header, data, signature)
+	if err == nil {
+		s.latestBlockDataHeight = header.Height()
```

Comment:
You would need a mutex to protect `latestBlockDataHeight` from concurrent read in `PruneBlockData`. An atomic type would work as well